fuzz/oss-fuzz-build.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash -eu
+
+TELEPORT_PREFIX="github.com/gravitational/teleport"
+
+prepare_teleport() {
+
+  go mod tidy
+
+  go get github.com/AdamKorcz/go-118-fuzz-build/utils
+  go get k8s.io/client-go/tools/clientcmd@v0.23.3
+  go get github.com/ThalesIgnite/crypto11@v1.2.4
```

Comment:
The first package is required by go oss-fuzz. I am not happy to include the specific versions too, `k8s.io/client-go` is needed for several fuzzers and even when for the `go test -fuzz` the dependencies are resolved automatically, oss-fuzz (that is based on the libfuzzer engine) reports these errors during my test:

```
replacing *testing.F
/root/go/pkg/mod/github.com/!thales!ignite/crypto11@v1.2.4/aead.go:29:2: missing go.sum entry for module providing package github.com/miekg/pkcs11 (imported by github.com/ThalesIgnite/crypto11); to add:
	go get github.com/ThalesIgnite/crypto11@v1.2.4
```

```
go: downloading github.com/jmespath/go-jmespath v0.4.0
go: downloading golang.org/x/text v0.3.7
go: downloading github.com/cenkalti/backoff/v4 v4.1.3
go: downloading github.com/aws/aws-sdk-go-v2/internal/endpoints/v2 v2.0.0
go: downloading github.com/kr/text v0.2.0
go: downloading github.com/rogpeppe/go-internal v1.8.0
go: downloading github.com/google/certificate-transparency-go v1.0.21
replacing *testing.F
# github.com/aws/aws-sdk-go-v2/internal/ini
/root/go/pkg/mod/github.com/aws/aws-sdk-go-v2/internal/ini@v1.3.0/fuzz.go:13:21: not enough arguments in call to Parse
	have (*bytes.Reader)
	want (io.Reader, string)
```

I was not able to find a better solution than including this minimal set of packages with specific versions. 

Since https://github.com/gravitational/teleport/pull/13473#discussion_r896870743 is related, I'll answer here, I noticed the similar workaround here https://github.com/google/oss-fuzz/blob/master/projects/teleport/build.sh with the `fuzz.go` removal.

I can try to debug it again tomorrow to understand why this is happening and come up with a better solution. Also, if you have any suggestions on why this occurs, let me please know.

Reply:
Update, to reproduce this, I cloned the teleport and did `go get github.com/AdamKorcz/go-118-fuzz-build/utils` (needed by oss-fuzz).

Then `go get` then upgraded some dependencies, including these below

```
-       github.com/imdario/mergo v0.3.5 // indirect
+       github.com/imdario/mergo v0.3.12 // indirect
-       github.com/miekg/pkcs11 v1.0.3-0.20190429190417-a667d056470f // indirect
+       github.com/miekg/pkcs11 v1.0.3 // indirect
```

I did not find a solution to avoid the upgrade, but what I can do is to update all dependencies at once, e.g., this works:

```sh
  go get -u all || true
  go mod tidy
  go get github.com/AdamKorcz/go-118-fuzz-build/utils
```

If you want to ask me about the `true` statement, it's because I prefer to run the shell script with the parameter to stop when error occurs and `go get -u all` at the end yields these, which is indeed evaluated by bash as error:

```
go: all: module github.com/Azure/azure-sdk-for-go/sdk/azcore@upgrade found (v1.1.0), but does not contain package github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/pipeline
go: all: module github.com/chai2010/gettext-go@upgrade found (v1.0.2), but does not contain package github.com/chai2010/gettext-go/gettext
go: all: module github.com/chai2010/gettext-go@upgrade found (v1.0.2), but does not contain package github.com/chai2010/gettext-go/gettext/mo
go: all: module github.com/chai2010/gettext-go@upgrade found (v1.0.2), but does not contain package github.com/chai2010/gettext-go/gettext/plural
go: all: module github.com/chai2010/gettext-go@upgrade found (v1.0.2), but does not contain package github.com/chai2010/gettext-go/gettext/po
go: all: module k8s.io/client-go@upgrade found (v0.24.1), but does not contain package k8s.io/client-go/pkg/apis/clientauthentication/v1alpha1
go: all: module sigs.k8s.io/kustomize/api@upgrade found (v0.11.5), but does not contain package sigs.k8s.io/kustomize/api/filters/prefixsuffix
```

Reply:
We can leave it for now, but it's a little concerning that the fuzzers will be running with different versions of the dependencies than what is actually used in the product and defined in go.mod.